rotkehlchen/db/dbhandler.py

```diff
@@ -223,6 +223,67 @@ def db_tuple_to_str(
     raise AssertionError('db_tuple_to_str() called with invalid tuple_type {tuple_type}')
 
 
+def combine_asset_balances(
+        asset_a_balances: List[SingleDBAssetBalance],
+        asset_b_balances: List[SingleDBAssetBalance],
+) -> List[SingleDBAssetBalance]:
+    """
+    Combine assets balances from two different queries. It iterates over the query results
+    of both queries and returns them sorted by timestamp combining the ones that share
+    timestamp and category
+    """
+    if len(asset_b_balances) == 0:
+        return asset_a_balances
+    if len(asset_a_balances) == 0:
+        return asset_b_balances
+
+    final_balances = []
+    # Consume first the earliest entry
+    earlier, latest = asset_a_balances, asset_b_balances
+    if earlier[0].time > latest[0].time:
+        earlier, latest = latest, earlier
+
+    latest_reading_index = 0
+    for balance in earlier:
```

Comment:
This entire function again seems way too complicated for what you are trying to do.

So what do we have? We have two lists of balances of two different assets. List 1 is sorted by timestamp, and list 2 the same.

We simply want to combine entries that have the same timestamp.

Why not:

1. Combine the lists into a single one
2. Sort it again by timestamp.
3. Create an empty result list
4. Iterate the combined list, consuming 2 items at a time moving items to the result list and summing them if the timestamp is the same.

The end result would be the same but at least if I understand your algorithm correctly it would be be much simpler and probably as performant.